### PR TITLE
Web – Delete documents

### PR DESCRIPTION
**Pull request by jcontonio**
* Adds delete action and ui
* Removes the separate list/documentsById in documents reducer, only using projectDocumentsById
* Changes babelrc targets to browsers, instead of Node. Was getting bizarre behavior with the destructuring in the reducer where documents wouldn't remove from state

### CODE REVIEW

File: web/src/client/js/actions/index.js [hunk 1]
```diff
@@ -15,10 +16,12 @@ export const ActionTypes = {
   // Documents
   REQUEST_DOCUMENTS: 'REQUEST_DOCUMENTS',
   RECEIVE_DOCUMENTS: 'RECEIVE_DOCUMENTS',
-  CREATE_DOCUMENT: 'CREATE_DOCUMENT',
-  UPDATE_DOCUMENT: 'UPDATE_DOCUMENT',
+  INITIATE_CREATE_DOCUMENT: 'INITIATE_CREATE_DOCUMENT',
```
Comment (jjidt) [inline web/src/client/js/actions/index.js]: kinda nitpicky, but the rest of the actions in this file have the resource before the action

so these would follow `INITIATE_DOCUMENT_CREATE`

File: web/src/client/js/actions/index.js [hunk 2]
```diff
@@ -15,10 +16,12 @@ export const ActionTypes = {
...
   RECEIVE_CREATED_DOCUMENT: 'RECEIVE_CREATED_DOCUMENT',
+  INITIATE_UPDATE_DOCUMENT: 'INITIATE_UPDATE_DOCUMENT',
   RECEIVE_UPDATED_DOCUMENT: 'RECEIVE_UPDATED_DOCUMENT',
+  INITIATE_DOCUMENT_REMOVE: 'INITIATE_DOCUMENT_REMOVE',
+  DOCUMENT_REMOVED: 'DOCUMENT_REMOVED',
```
Comment (jjidt) [inline web/src/client/js/actions/index.js]: `REMOVE_DOCUMENT` for consistency

File: web/src/client/js/actions/index.js [hunk 3]
```diff
@@ -15,10 +16,12 @@ export const ActionTypes = {
   // Documents
   REQUEST_DOCUMENTS: 'REQUEST_DOCUMENTS',
   RECEIVE_DOCUMENTS: 'RECEIVE_DOCUMENTS',
-  CREATE_DOCUMENT: 'CREATE_DOCUMENT',
-  UPDATE_DOCUMENT: 'UPDATE_DOCUMENT',
+  INITIATE_CREATE_DOCUMENT: 'INITIATE_CREATE_DOCUMENT',
   RECEIVE_CREATED_DOCUMENT: 'RECEIVE_CREATED_DOCUMENT',
+  INITIATE_UPDATE_DOCUMENT: 'INITIATE_UPDATE_DOCUMENT',
```
Comment (jjidt) [inline web/src/client/js/actions/index.js]: same

**Review by jjidt — DISMISSED** [review]
Everything is looking really nice, just the one thing on the action names

**Review by dheniges — CHANGES_REQUESTED** [review]
There's an issue with document update now, if you change the doc title the reducer throws an error when trying to update the doc.

On a side note, I think we need two babel configs, one for the web server, and one for the web client / webpack build. Since chrome is a target, I assume v8 should handle whatever chrome output and it seems to work. But it seems like we should be targeting node for the server. I'll open a separate issue for this.

File: web/src/client/js/components/Document/DocumentComponent.js [hunk 4]
```diff
@@ -1,26 +1,54 @@
...
+  }
+  const changeHandlerAction = debounce(500, (e) => {
+    nameChangeHandler(e)
+  })
   return (
     <div className="document" key={docKey}>
       <header>
-        <textarea
+        <div
+          contentEditable
```
Comment (dheniges) [inline web/src/client/js/components/Document/DocumentComponent.js]: according to MDN docs, this should explicitly be set to "true" https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable

Comment (dheniges) [reply web/src/client/js/components/Document/DocumentComponent.js]: Also, is contenteditable the right tool for this? It's going to insert html entities and elements. If you put a few spaces in it inserts `&nbsp;`, which we don't want. The return key is overridden on this element, but if it's not hitting return inserts div elements with contenteditable.

**Review by dheniges — CHANGES_REQUESTED** [review]
questions on contenteditable

File: web/src/client/js/components/Document/DocumentComponent.js (hunk 4 above)
Comment (jcontonio) [reply web/src/client/js/components/Document/DocumentComponent.js]: Re: contenteditable -> I had previously used textarea, which worked nicely, except that you can't style a textarea to line up with anything. Let me try some more things and see if I can get back to a textarea.

Note that documentList is also using contenteditable.

**Review by dheniges — APPROVED** [review]

**Pull request merged**